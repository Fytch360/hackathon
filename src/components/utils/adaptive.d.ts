declare module '@sbol/design-system/core/utils/adaptive' {
    // @types/lodash
    interface MapCache {
        delete(key: unknown): boolean
        get(key: unknown): unknown
        has(key: unknown): boolean
        set(key: unknown, value: unknown): this
        clear?: (() => void) | undefined
    }

    interface MemoizedFunction {
        cache: MapCache;
    }

    export const isTouchable: (() => boolean) & MemoizedFunction
    export const isNotTouchable: (() => boolean) & MemoizedFunction
    export const applyForTouchable: (value: string, defaultValue: string) => string

    /* Mobile | Desktop */
    export const isMobilePlatform: (() => boolean) & MemoizedFunction
    export const isMobile: (() => boolean) & MemoizedFunction
    export const isDesktopPlatform: (() => boolean) & MemoizedFunction

    /* IE */
    export const isIE: boolean
}
