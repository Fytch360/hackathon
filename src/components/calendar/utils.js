import {
    startOfWeek,
    endOfWeek,
    startOfMonth,
    endOfMonth,
    startOfYear,
    endOfYear,
    eachDayOfInterval,
    eachWeekOfInterval,
    eachMonthOfInterval,
    isSameMonth,
    startOfDecade,
    addYears,
    setMinutes,
    setHours,
    parseISO,
    formatISO,
    parse,
    isWithinInterval,
    format as formatDate,
    isBefore,
    isSameDay
} from 'date-fns'
import { chunk } from 'lodash'

import {
    locale,
    MIN_YEAR_RESTRICTION_DATE,
    MAX_YEAR_RESTRICTION_DATE,
    MONTHS_IN_ROW,
    YEARS_IN_ROW,
    YEARS_IN_DECADE,
    DATE_DEFAULT_HOUR,
    DATE_DEFAULT_MINUTE,
    commonMaskOptions,
    MaskFormat,
    MaskPattern,
    InputMode
} from './constants'

const options = {
    locale,
}

export const setDefaultHHMM = (date) =>
    setMinutes(setHours(date, DATE_DEFAULT_HOUR), DATE_DEFAULT_MINUTE)
export const capitalize = (str) => str.charAt(0).toUpperCase() + str.slice(1)

// Возвращает массив массивов из дней для указанной даты
export const getMonthDates = (date, viewModeLight) => {
    const dateParsed = parseISO(date)
    const monthDates = []

    const weeks = eachWeekOfInterval(
        {
            start: startOfMonth(dateParsed),
            end: endOfMonth(dateParsed),
        },
        options
    )

    weeks.forEach((week, weekIndex) => {
        monthDates[weekIndex] = []

        const days = eachDayOfInterval({
            start: startOfWeek(week, options),
            end: endOfWeek(week, options),
        })

        days.forEach((day, dayIndex) => {
            monthDates[weekIndex][dayIndex] = isSameMonth(dateParsed, day)
                ? formatISO(setDefaultHHMM(day))
                : null
        })
    })

    return monthDates
}

// Возвращает массив массивов из месяцев для указанной даты
export const getYearDates = (date) => {
    const dateParsed = parseISO(date)
    const start = startOfYear(dateParsed)
    const end = endOfYear(dateParsed)
    const yearDates = eachMonthOfInterval({ start, end }).map((month) =>
        formatISO(setDefaultHHMM(month))
    )

    return chunk(yearDates, MONTHS_IN_ROW)
}

// Возвращает массив массивов из десятилетия для указанной даты
export const getDecadeDates = (date) => {
    const dateParsed = parseISO(date)
    const decadeDates = new Array(YEARS_IN_DECADE).fill().map((_, index) => {
        const startDateOfDecade = startOfDecade(dateParsed)
        const nextDateOfDecade = addYears(
            startDateOfDecade,
            parseInt(index, 10)
        )

        return formatISO(setDefaultHHMM(nextDateOfDecade))
    })

    return chunk(decadeDates, YEARS_IN_ROW)
}

export const createMaskOptions = (mode, restriction, mask) => ({
    ...commonMaskOptions,
    pattern: mask || MaskPattern[mode],
    min: Array.isArray(restriction) || !restriction.start ? commonMaskOptions.min : restriction.start,
    max: Array.isArray(restriction) || !restriction.end ? commonMaskOptions.max : restriction.end,
    format: (date) => formatDate(parseISO(date), MaskFormat[mode], { locale }),
    parse: (string) => {
        if (mode === InputMode.DEFAULT) {
            const [day, month, year] = string.split('.')
            return new Date(year, month - 1, day)
        }

        return parse(mode === InputMode.DAY ? string.match(/\d+/g).pop() : string, MaskFormat[mode], new Date(), { locale })
    }
})

export const getInitialError = (date, restriction, validation) => {
    if (!date) {
        return ''
    }

    const dateParsed = parseISO(date)
    const error = Array.isArray(restriction) ? !restriction.some((available) => validation(available, dateParsed)) : !isWithinInterval(dateParsed, restriction)
    
    return error ? ' ' : ''
}

export const getInitialRestriction = (restriction) => {
    if (Array.isArray(restriction) && restriction.length) {
        return restriction
    }

    return {
        start: restriction.start || MIN_YEAR_RESTRICTION_DATE,
        end: restriction.end || MAX_YEAR_RESTRICTION_DATE
    }
}

export const getBackgroundColor = ({ theme, isStartDate, isEndDate, disabled }) => {
    if ((isStartDate || isEndDate) && !disabled) {
        return 'transparent'
    }

    if (disabled) {
        return theme.calendarButtonInvalid
    }

    return theme.calendarButtonWithin
}

export const swapDates = (state) => {
    const { start, end } = state

    // Если дата конца раньше чем дата начала,
    // то свапаем даты местами
    if (isBefore(parseISO(end), parseISO(start))) {
        state.start = end
        state.end = start
    }
}
