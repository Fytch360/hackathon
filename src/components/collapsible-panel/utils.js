import React from 'react'

const passProps = (children, props) =>
    typeof children.type === 'function' ? React.cloneElement(children, { ...props }) : children

/**
 * @param {function, node} children
 * @param {object} props, что будут прокинуты внутрь children, если он является ReactNode(function), иначе вернет как есть
 * @returns {function, node} всё то же самое, только с прокинутыми props (по условию выше)
 */

export const passPropsToChildren = (children, props) =>
    children && children.map
        ? children.map((item, index) => passProps(item, { key: index, ...props }))
        : passProps(children, { ...props })
