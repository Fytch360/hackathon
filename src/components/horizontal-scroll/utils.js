export const easeInOutQuad = (time, start, change, duration) => {
    // comment: лучше не скомпоновалось
    /* eslint-disable no-mixed-operators */
    const half = 2
    let halvedTime = time / (duration / half)
    if (halvedTime < 1) {
        return (change / half) * halvedTime * halvedTime + start
    }
    halvedTime -= 1
    return -change / half * (halvedTime * (halvedTime - half) - 1) + start
}

export const smoothScroll = (element, change, duration, cb) => {
    const start = element.scrollLeft
    const el = element
    let currentTime = 0
    const increment = 20

    const animateScroll = () => {
        currentTime += increment
        el.scrollLeft = easeInOutQuad(currentTime, start, change, duration)
        if (currentTime < duration) {
            requestAnimationFrame(animateScroll)
        } else {
            cb()
        }
    }
    animateScroll()
}

export const leftCheck = (scroll) => {
    if (!scroll) {
        return false
    }

    return scroll.scrollLeft > 0
}

export const rightCheck = (scroll) => {
    if (!scroll) {
        return false
    }

    // В IE scrollWidth больше clientWidth на 1px, даже если нет скролла, поэтому  > 1, а не > 0.
    return Math.floor(scroll.scrollWidth - scroll.clientWidth - scroll.scrollLeft) > 1
}

export const scrollToChild = (scroll, target, duration, cb) => {
    const scrollViewPortSize = scroll.clientWidth
    const targetSize = target.scrollWidth
    const scrollDistance = scroll.scrollLeft
    const targetScrollOffset = target.offsetLeft
    
    const scrollEdge = (scrollDistance < targetScrollOffset) && (targetSize < scrollViewPortSize) ? 'right' : 'left'

    const targetOffset = target.getBoundingClientRect()?.[scrollEdge]
    const scrollOffset = scroll.getBoundingClientRect()?.[scrollEdge]

    const distanceToTarget = targetOffset - scrollOffset

    smoothScroll(scroll, distanceToTarget, duration, cb)
}
