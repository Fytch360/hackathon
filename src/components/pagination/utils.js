import { useState } from 'react'

const PAGE_2 = 2
const PAGE_4 = 4
const PAGE_5 = 5

export const getMediaListeners = (media, cb) => {
    if (media.addEventListener) {
        return {
            addMediaListener: () => media.addEventListener('change', cb),
            removeMediaListener: () => media.removeEventListener('change', cb)
        }
    }

    // https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/addListener
    // backwards compatibility IE, Opera
    return {
        addMediaListener: () => media.addListener(cb),
        removeMediaListener: () => media.removeListener(cb)
    }
}

export const usePagination = ({
    count = 1,
    defaultPage = 1,
    page,
    onChange: handleChange,
    mobile = false
}) => {
    const [pageState, setPageState] = useState(page || defaultPage)

    const handleClick = (event, value) => {
        setPageState(value)
        if (handleChange) {
            handleChange(value)
        }
    }

    const start = Math.max(
        Math.min(
            // Если по центру нужно с -1 элемента начать
            pageState > PAGE_5 ? pageState - 1 : 1,
            // Первая цифра за троеточием в конце
            count - PAGE_4,
        ),
        // Первая цифра перед троеточием в начале
        PAGE_2,
        // Если в конце списка, то значение count - 1
        pageState > count - PAGE_5 ? count - PAGE_4 : 1,
    )

    const end = Math.min(
        Math.max(
            // Если по центру, то +1 элемент рисуем
            pageState > PAGE_5 ? pageState + 1 : 1,
            // Если в конце списка, то значение count - 1
            pageState > count - PAGE_5 ? count - 1 : 1,
            // Последняя цифра за троеточием в начале
            PAGE_5,
        ),
        // Последняя цифра после троеточия в конце
        count - 1,
    )

    // Создаем массив с числами от start до end
    // https://dev.to/namirsab/comment/2050
    const range = () => {
        const length = end - start + 1
        return length >= 0 && Array.from({ length }, (_, i) => start + i)
    }

    const mobileRange = () => {
        const length = count < 3 ? count : 3
        const page = Math.max(
            count === pageState ? pageState - 2 : pageState - 1,
            1,
        )
        return length >= 0 && Array.from({ length }, (_, i) => page + i)
    }

    const rangeArr = range()
    const mobileRangeArr = mobileRange()

    const itemList = mobile ? [
        count > 1 ? 'first' : [],
        // Кнопка назад
        count > 1 ? 'previous' : [],
        ...mobileRangeArr,
        count > 1 ? 'next' : [],
    ] :
        [
        // Кнопка назад
            count > 1 ? 'previous' : [],
            // 1 страница
            1,
            // Вставляем многоточие, если нужная страница не первые пять
            ...(start >= PAGE_5 ? ['doublePrevious'] : []),
            // Вставляем середину страницы которую должно быть видно
            ...(rangeArr || []),
            // Вставляем многоточие, если нужная страница не последние пять
            ...(end <= count - PAGE_4 ? ['doubleNext'] : []),
            // Последняя страница
            count > 1 ? count : [],
            count > 1 ? 'next' : [],
        ]

    const buttonPage = (type) => {
        switch (type) {
            case 'doublePrevious':
                return pageState - 3
            case 'previous':
                return pageState - 1
            case 'next':
                return pageState + 1
            case 'doubleNext':
                return pageState + 3
            case 'first':
                return 1
            default:
                return null
        }
    }

    const items = itemList.map((item) => typeof item === 'number'
        ? {
            onClick: (event) => {
                handleClick(event, item)
            },
            type: 'page',
            pageItem: item,
            selected: item === pageState,
            'aria-current': item === pageState ? 'true' : 'false',
        }
        : {
            onClick: (event) => {
                handleClick(event, buttonPage(item))
            },
            type: item,
            pageItem: buttonPage(item),
            selected: false,
            disabled:
            (item.indexOf('previous') === -1 || item.indexOf('next') === -1 || item.indexOf('first') === -1) &&
              (item === 'next' ? pageState >= count : pageState <= 1),
        })


    return { items }
}
