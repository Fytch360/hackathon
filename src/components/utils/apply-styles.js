// comment: for demo purpose
/* eslint-disable valid-jsdoc */
function applyStyles (styles, props) {
    // comment: more visual
    return typeof styles === 'function'
        ? styles(props)
        : Array.isArray(styles)
            ? styles.map((s) => applyStyles(s, props))
            : styles
}

/**
 * Makes it easier to create an emotion component which
 * accepts props to enable/disable certain styles.
 *
 * accepts an object where the key is a prop and the value
 * is the styles that should be applied if that prop is
 * passed. Returns a function which you pass to a
 * emotionComponentFactory.
 *
 * @param {Object} styles The prop to styles object
 * @return {Function} the dynamic styles function
 * @example
 *
 * export const Text = styled(Box)(
    propStyles({
        nowrap: {
            whiteSpace: 'nowrap'
        },
        ellipsis: {
            whiteSpace: 'nowrap',
            overflow: 'hidden',
            textOverflow: 'ellipsis'
        },
        align: css`
            text-align: ${(p) => p.align};
        `,
        bold: {
            fontWeight: 'bold'
        },
        uppercase: {
            textTransform: 'uppercase'
        }
    })
)
 */
export function propStyles (styles) {
    return function dynamicStyles (props) {
        return Object.keys(props).map((key) => {
            if (styles[key] && props[key]) {
                return applyStyles(styles[key], props)
            }
            return null
        })
    }
}

/**
 * Makes it easier to create an emotion component
 * which accepts enums for certain variants of styles
 *
 * Accepts an object where the key is a variant name
 * (the prop consumers will use) and the value is an
 * object where those keys are the possible values for
 * the variant prop, and the value is the styles to be
 * applied.
 *
 * @example
 const Text = styled(Box)(
    background: '#fff';
    variantStyles({
        variant: {
            headline: {
                fontSize: '32px',
                lineHeight: '1.125'
            },
            title: {
                fontSize: '22px',
                lineHeight: '1.1818'
            },
            body: {
                fontSize: '16px',
                lineHeight: '1.25'
            },

            caption: {
                fontSize: '14px',
                lineHeight: '1.2857'
            }
        }
    })
)
 */
export function variantStyles (styles) {
    return function dynamicStyles (props) {
        return Object.entries(props).map(([key, value]) => {
            if (styles[key]) {
                return applyStyles(styles[key][value], props)
            }
            return null
        })
    }
}
