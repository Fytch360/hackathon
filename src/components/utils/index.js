export { formatNumber } from './format-number'
export { makeMoneyMask } from './make-money-mask'
export { formatValueByMask } from './format-value'
export { formatLocalPhone } from './format-local-phone'
export { propStyles, variantStyles } from './apply-styles'
export { scrollToElement } from './scroll-to'
export { getFilteredComponent } from './get-filtered-component'
export { getDisplayName } from './get-display-name'
