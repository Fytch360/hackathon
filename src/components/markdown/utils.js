// comment: кейсинг предоставляет Remarkable
/* eslint-disable camelcase */
import _ from 'lodash'
import Remarkable from 'remarkable'
import { escapeHtml, replaceEntities } from 'remarkable/lib/common/utils'

import { externalRegular as ExternalIcon } from '../icon/common'

import { textColorParcer, textColorRule } from './text-color'

const options = {
    html: false,
    xhtmlOut: false,
    breaks: false,
    langPrefix: 'language-',
    linkify: false,
    typographer: true,
    quotes: '«»‘’'
}

const internalTestRegExp = /^[#./]/
const lastWorldRegExp = /\S*$/
const nbspRegExp = /\u00A0/g
const nbspHtmlEntity = '&nbsp;'

export const isExternalUrl = (url = '') =>
    !!url && !internalTestRegExp.test(url)

export const linkOpen = (tokens, idx) => {
    const href = escapeHtml(tokens[idx].href)
    const isExternal = isExternalUrl(href)
    const title = tokens[idx].title
        ? ` title="${escapeHtml(replaceEntities(tokens[idx].title))}"`
        : ''

    if (!isExternal) {
        return `<a href="${href}"${title}><span>`
    }

    const externals = ' target="_blank" rel="noopener noreferrer"'
    const content = escapeHtml(tokens[idx + 1].content)
    const droppedLast = content.replace(lastWorldRegExp, '')

    return `<a href="${href}"${title}${externals}>${droppedLast ? '<span>' : ''}`
}

export const linkClose = (tokens, idx) => {
    // comment: приходят все токены, а href лежит в два токена слева: <a> и текст
    const href = escapeHtml(tokens[idx - 2].href)
    if (!isExternalUrl(href)) {
        return '</span></a>'
    }

    const content = escapeHtml(tokens[idx - 1].content)
    const droppedLast = content.replace(lastWorldRegExp, '')
    const lastWord = content.match(lastWorldRegExp)[0]

    return `${droppedLast ? '</span>' : ''}<span>${lastWord}${ExternalIcon}</span></a>`
}

export const text = (tokens, idx) => {
    const content = escapeHtml(tokens[idx].content)
    const href = escapeHtml(tokens[idx - 1]?.href)

    if (!isExternalUrl(href)) {
        return content
    }

    return content.replace(lastWorldRegExp, '')
}

export const replaceNbsp = (stateCore) => {
    stateCore.src = stateCore.src?.replace(nbspRegExp, nbspHtmlEntity)
}

const inlineDisable = [
    'backticks',
    'ins',
    'mark',
    'footnote_inline',
    'footnote_ref'
]
const blockDisable = ['blockquote', 'code', 'fences']
const blockDisableAdditional = ['deflist', 'table']

const configMdCommon = () => {
    const markdown = new Remarkable('full', options)

    markdown.core.ruler.before('block', 'nbspReplacer', replaceNbsp)
    markdown.inline.ruler.push('text_color', textColorRule)
    markdown.inline.ruler.disable(inlineDisable)
    markdown.renderer.rules.link_open = linkOpen
    markdown.renderer.rules.link_close = linkClose
    markdown.renderer.rules.text = text
    markdown.renderer.rules.color = textColorParcer

    return markdown
}

const configMdFull = () => {
    const markdownFull = configMdCommon()
    markdownFull.block.ruler.disable(blockDisable)

    return markdownFull
}

const configMdShort = () => {
    const markdownShort = configMdCommon()
    // comment: нужно поменять,но убедится что ничего не сломалось
    markdownShort.block.ruler.disable(blockDisable.concat(blockDisableAdditional))

    // ignores images completely
    markdownShort.renderer.rules.image = _.stubString

    return markdownShort
}

export const markdownFull = configMdFull()
export const markdownShort = configMdShort()
