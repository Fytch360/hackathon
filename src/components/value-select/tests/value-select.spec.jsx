import React from 'react'
import { mount } from 'enzyme'

import { ValueSelect, ValueOption } from '..'
import { keyCodes } from '../utils'

describe('<Autocomplete />', () => {
    const optionsExample = [
        { title: 'Title 0', value: 'Value 0' },
        { title: 'Title 1', value: 'Value 1' },
        { title: 'Title 2', value: 'Value 2' },
    ]
    const propsExample = {
        id: 'example-id',
        size: 'lg',
        mode: 'select',
        translations: {
            placeholder: 'placeholderExample'
        }
    }
    let wrapper
    let onChangeMock

    beforeEach(() => {
        onChangeMock = jest.fn()

        wrapper = mount(
            <ValueSelect {...propsExample} onChange={onChangeMock} >
                {optionsExample.map(({ title, value }) =>
                    <ValueOption key={value} value={value} title={title} />
                )}
            </ValueSelect>
        )
    })

    test('Should be available', () => {
        expect(ValueSelect).toBeDefined()
    })

    test('Snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('Should\'nt render Contents by default', () => {
        expect(wrapper.find('ContentsStyled').exists()).toBeFalsy()
    })

    test('Should render Contents when component was focussed', () => {
        wrapper.find('TargetStyled')
            .simulate('click', { })

        expect(wrapper.find('ContentsStyled')).toHaveLength(1)
    })

    test('Should return first option value if pressed only enter key', () => {
        wrapper.find('TargetStyled')
            .simulate('click', { })
        
        wrapper.find('ContentsViewStyled')
            .simulate('keydown', { keyCode: keyCodes.KEY_ENTER })

        expect(onChangeMock.mock.calls[0][0]).toBe(optionsExample[0].value)
    })

    test('Should return first option value if pressed only space key', () => {
        wrapper.find('TargetStyled')
            .simulate('click', { })
        
        wrapper.find('ContentsViewStyled')
            .simulate('keydown', { keyCode: keyCodes.KEY_SPACE })

        expect(onChangeMock.mock.calls[0][0]).toBe(optionsExample[0].value)
    })

    test('Should return first option value if gets autoselect mode', () => {
        wrapper.setProps({ mode: 'autoselect' })

        expect(onChangeMock.mock.calls[0][0]).toBe(optionsExample[0].value)
    })

    test('Should return first option\'s value if pressed home key and then option was choosed', () => {
        wrapper.find('TargetStyled')
            .simulate('click', { })
        
        wrapper.find('ContentsViewStyled')
            .simulate('keydown', { keyCode: keyCodes.KEY_HOME })
            .simulate('keydown', { keyCode: keyCodes.KEY_ENTER })
        
        expect(onChangeMock.mock.calls[0][0]).toBe(optionsExample[0].value)
    })

    test('Should return last option\'s value if pressed end key and then option was choosed', () => {
        const lastOptionIndex = optionsExample.length - 1

        wrapper.find('TargetStyled')
            .simulate('click', { })
        
        wrapper.find('ContentsViewStyled')
            .simulate('keydown', { keyCode: keyCodes.KEY_END })
            .simulate('keydown', { keyCode: keyCodes.KEY_ENTER })
        
        expect(onChangeMock.mock.calls[0][0]).toBe(optionsExample[lastOptionIndex].value)
    })

    test('Should return second option\'s value if pressed arrow down and then option was choosed', () => {
        wrapper.find('TargetStyled')
            .simulate('click', { })
        
        wrapper.find('ContentsViewStyled')
            .simulate('keydown', { keyCode: keyCodes.KEY_ARROW_DOWN })
            .simulate('keydown', { keyCode: keyCodes.KEY_ENTER })
        
        expect(onChangeMock.mock.calls[0][0]).toBe(optionsExample[1].value)
    })

    test('Should return penultimate option\'s value if pressed end key then pressed arrow up and then option was choosed', () => {
        const lastOptionIndex = optionsExample.length - 1

        wrapper.find('TargetStyled')
            .simulate('click', { })
        
        wrapper.find('ContentsViewStyled')
            .simulate('keydown', { keyCode: keyCodes.KEY_END })
            .simulate('keydown', { keyCode: keyCodes.KEY_ARROW_UP })
            .simulate('keydown', { keyCode: keyCodes.KEY_ENTER })
        
        expect(onChangeMock.mock.calls[0][0]).toBe(optionsExample[lastOptionIndex - 1].value)
    })

    test('Should\'nt render Contents if Tab key was pressed', () => {
        wrapper.find('TargetStyled')
            .simulate('click', { })
        
        wrapper.find('ContentsViewStyled')
            .simulate('keydown', { keyCode: keyCodes.KEY_TAB })
        
        expect(wrapper.find('ContentsStyled').exists()).toBeFalsy()
    })

    test('Should\'nt render Contents if esc key was pressed', () => {
        wrapper.find('TargetStyled')
            .simulate('click', { })
        
        wrapper.find('ContentsViewStyled')
            .simulate('keydown', { keyCode: keyCodes.KEY_ESCAPE })
        
        expect(wrapper.find('ContentsStyled').exists()).toBeFalsy()
    })

    test('Should render MultiSelectedOptions if gets multiselect mode', () => {
        wrapper.setProps({ mode: 'multiselect', value: ['Value 0', 'Value 1'] })

        expect(wrapper.find('MultiSelectedOptions')).toHaveLength(1)
    })

    test('Should\'nt call onChange if pressed ctrl + a w/o multiselect mode', () => {
        wrapper.find('TargetStyled')
            .simulate('click', { })
        
        wrapper.find('ContentsViewStyled')
            .simulate('keydown', { keyCode: keyCodes.KEY_A, ctrlKey: true })

        expect(onChangeMock.mock.calls).toHaveLength(0)
    })

    test('Should return array w/ all options values if pressed ctrl + a w/ multiselect mode', () => {
        wrapper.setProps({ mode: 'multiselect', value: ['Value 0'] })

        wrapper.find('TargetStyled')
            .simulate('click', { })
        
        wrapper.find('ContentsViewStyled')
            .simulate('keydown', { keyCode: keyCodes.KEY_A, ctrlKey: true })

        expect(onChangeMock.mock.calls[0][0]).toEqual(
            optionsExample.map(({ value }) => value)
        )
    })
})
