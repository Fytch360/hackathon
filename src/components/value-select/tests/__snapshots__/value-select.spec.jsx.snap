// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Autocomplete /> Snapshot 1`] = `
.emotion-0 {
  position: relative;
}

.emotion-2 {
  -webkit-tap-highlight-color: transparent;
  outline: none;
  border-radius: 4px;
  border: solid 1px;
  width: 100%;
  text-align: left;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  padding: 0;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-right: 16px;
  padding-left: 16px;
  padding-right: calc(26px + 20px);
  -webkit-transition: 0.17s ease box-shadow;
  transition: 0.17s ease box-shadow;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  min-height: calc(64px + 2px);
}

.emotion-2::after {
  content: '';
  min-height: calc(64px - 2*12px);
  font-size: 0;
}

.emotion-2:hover {
  border: solid 1px;
}

.emotion-2:active {
  border: solid 1px;
}

body:not(.pointer-events) .emotion-2:focus {
  -webkit-transition: 0.17s ease;
  transition: 0.17s ease;
  box-shadow: inset 0px 0px 0px 1px;
}

.emotion-4 {
  letter-spacing: -0.3px;
  text-align: left;
  font-weight: 500;
  font-size: 19px;
  line-height: 24px;
  margin: 0;
  padding: 0;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-6 {
  position: absolute;
  right: 9px;
  top: 50%;
  -webkit-transform: translate(0, -50%);
  -moz-transform: translate(0, -50%);
  -ms-transform: translate(0, -50%);
  transform: translate(0, -50%);
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
}

.emotion-8 {
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
}

.emotion-8 svg {
  display: block;
}

<ValueSelect
  id="example-id"
  mode="select"
  onChange={[MockFunction]}
  size="lg"
  translations={
    Object {
      "placeholder": "placeholderExample",
    }
  }
>
  <OnClickOutside(Perimeter)
    disableOnClickOutside={true}
    eventTypes={
      Array [
        "mousedown",
        "touchstart",
      ]
    }
    excludeScrollbar={false}
    onClickOutside={[Function]}
    outsideClickIgnoreClass="ignore-react-onclickoutside"
    preventDefault={false}
    stopPropagation={false}
  >
    <Perimeter
      disableOnClickOutside={[Function]}
      enableOnClickOutside={[Function]}
      eventTypes={
        Array [
          "mousedown",
          "touchstart",
        ]
      }
      onClickOutside={[Function]}
      outsideClickIgnoreClass="ignore-react-onclickoutside"
      preventDefault={false}
      stopPropagation={false}
    >
      <WrapperStyled
        id="example-id-wrapper"
      >
        <div
          className="emotion-0 emotion-1"
          id="example-id-wrapper"
        >
          <TargetStyled
            aria-haspopup="listbox"
            aria-label=""
            aria-labelledby="example-id"
            disabled={false}
            id="example-id"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            open={null}
            readonly={false}
            size="lg"
            type="button"
          >
            <button
              aria-haspopup="listbox"
              aria-label=""
              aria-labelledby="example-id"
              className="emotion-2 emotion-3"
              disabled={false}
              id="example-id"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              open={null}
              size="lg"
              type="button"
            >
              <ItemNotChosenStyled
                fontWeight="medium"
                size="lg"
              >
                <p
                  className="emotion-4 emotion-5"
                  fontWeight="medium"
                  size="lg"
                >
                  placeholderExample
                </p>
              </ItemNotChosenStyled>
              <ArrowStyled
                open={null}
              >
                <div
                  className="emotion-6 emotion-7"
                  open={null}
                >
                  <Icon
                    colorScheme="primary"
                    icon="<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\">    <path d=\\"M15.293 9.883c.39-.39 1.024-.39 1.414 0 .39.39.39 1.024 0 1.414L12 16.004l-4.707-4.707c-.39-.39-.39-1.024 0-1.414.39-.39 1.024-.39 1.414 0L12 13.176l3.293-3.293z\\"/></svg>"
                  >
                    <WrapperStyled
                      colorScheme="primary"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\">    <path d=\\"M15.293 9.883c.39-.39 1.024-.39 1.414 0 .39.39.39 1.024 0 1.414L12 16.004l-4.707-4.707c-.39-.39-.39-1.024 0-1.414.39-.39 1.024-.39 1.414 0L12 13.176l3.293-3.293z\\"/></svg>",
                        }
                      }
                      onClick={[Function]}
                    >
                      <span
                        className="emotion-8 emotion-9"
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\">    <path d=\\"M15.293 9.883c.39-.39 1.024-.39 1.414 0 .39.39.39 1.024 0 1.414L12 16.004l-4.707-4.707c-.39-.39-.39-1.024 0-1.414.39-.39 1.024-.39 1.414 0L12 13.176l3.293-3.293z\\"/></svg>",
                          }
                        }
                        onClick={[Function]}
                      />
                    </WrapperStyled>
                  </Icon>
                </div>
              </ArrowStyled>
            </button>
          </TargetStyled>
        </div>
      </WrapperStyled>
    </Perimeter>
  </OnClickOutside(Perimeter)>
</ValueSelect>
`;
